docs/DeveloperGuide.adoc

```diff
@@ -295,71 +316,174 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
-_{More to be added}_
+|`* *` |coding student|share coding challenges with my friends easily|discuss possible solutions with them
+|`* *` |achievement hunter|view the badges that I have earned (and those that I have not)|feel a sense of accomplishment
+|`* *` |conscientious coding student|easily identify problems that I have given up on previously|tackle them again
+|`* *` |professor teaching this course|assign a unique hash code for each problem I input|students can look for the problems quickly
+
+|`* *` |picky coder|select different themes for the software|the user interface looks more appealing to me
+
+|`* *` |programming language polyglot|specifically choose problems designed in a specific language|practice that language in focus
+
+|`* *` |coding student|view similar/related problems to the one I have just completed |further deepen my understanding of the concepts used
+
+|`* *` |easily distracted coder|switch off all external distractions|focus better on the problem I am working on
+
+|`* *` |busy coder|save my progress on a problem|come back to it and continue at a later time
+
+|`* *` |competitive programming enthusiast|set my own questions and pose them to my fellow enthusiast friends to solve|
+
+|`* *` | student |see statistics about my attempts/success rates to track my learning progress|
+
+|`*` |tutor|print a pdf version of the coding problem |give them as practices to my students
+
+|`*` |competitive coder|see my areas for improvement after completing a coding challenge|become a better competitive coder
+
+|`*` |international student|view translation of the problem statement|aids my understanding of the problem
+|=======================================================================
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `Duke Academy` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
-
+=== Use case: UC01 Set questions
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to input problem sets.
+2. Duke Academy requires a file path.
+3. User select file path.
+4. Duke Academy imports the problem sets and prompts success message.
 +
 Use case ends.
 
 *Extensions*
+[none]
+* 4a. The input format is incorrect.
++
+Duke Academy reports wrong format error. Duke Academy resumes at step 3.
 
+=== Use case: UC02 View Questions
+*MSS*
+
+1. User finds a question by question ID, title or category.
+2. Duke Academy shows a list of problems that matches the keyword.
+3. User views the question identified by ID.
+4. Duke Academy displays the problem statement of the question.
++
+Use case ends.
+
+*Extensions*
 [none]
-* 2a. The list is empty.
+* 1a. User inputs wrong keywords.
```

Comment:
I think instead of 'user inputs wrong keywords', it should be 'user's inputted keyword cannot be found' because technically I think the keyword cannot be wrong since at most is the keyword cannot be found, so maybe can show a "no question found" message if this occurs.